### PR TITLE
(DOCSP-27005): @realm/react-ify Manage Email/Password Users page

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

@realm/react-ify Manage Email/Password Users page

> **Note**
>
> Had to update some dependencies to make tests pass. Don't fully understand why. RN testing confuses me.

### Jira

- https://jira.mongodb.org/browse/DOCSP-27005

### Staged Changes

- [Manage Email Password Users (RN)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27005/sdk/react-native/manage-users/manage-email-password-users/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/manage-users/manage-email-password-users.txt [hunk 1]
```diff
@@ -10,41 +10,55 @@ Manage Email/Password Users - React Native SDK
...
+---------------------------
+
+To register a new email/password user, pass the user's email address and desired
+password to :js-sdk:`EmailPasswordAuth.registerUser()
+<Realm.Auth.EmailPasswordAuth.html#registerUser>`. The email address must not be
+associated with another email/password user and the password must be between 6
+and 128 characters.
 
-.. note:: Confirm New Users
+You must :ref:`confirm a new user's email address <react-native-confirm-user>`
```
Comment (nlarew) [inline source/sdk/react-native/manage-users/manage-email-password-users.txt:57]:
```suggestion
After registration, you must :ref:`confirm a new user's email address <react-native-confirm-user>`
```

small clarification

File: source/sdk/react-native/manage-users/manage-email-password-users.txt [hunk 2]
```diff
@@ -76,22 +90,8 @@ To confirm a registered user, pass a valid ``token`` and ``tokenId`` to
...
-         await app.emailPasswordAuth.confirmUser({ token, tokenId });
-
-   .. tab::
-      :tabid: typescript
-
-      .. code-block:: typescript
-
-         await app.emailPasswordAuth.confirmUser({ token, tokenId });
-
+.. literalinclude:: /examples/generated/node/authenticate.snippet.confirm-email-pass-user.js
```
Comment (nlarew) [inline source/sdk/react-native/manage-users/manage-email-password-users.txt:93]: Can we remove the "Complete a User Confirmation" subheading? It's the only subsection and kind awkwardly breaks apart the content IMO.

Maybe add a sentence somewhere to replace it that mentions "the confirmation process starts when you register a user and ends when you confirm them from your client code"

**Review by nlarew — COMMENTED** [review]
2 small things otherwise LGTM

**Review by nlarew — APPROVED** [review]

**Review by nlarew — COMMENTED** [review]
Actually one more thing - looks like there's a bunch of unused files. Can we clean that up?

**Comment by mongodben** [comment]
> Actually one more thing - looks like there's a bunch of unused files. Can we clean that up?

going to deal w all of that in [DOCSP-27996](https://jira.mongodb.org/browse/DOCSP-27996)

File: examples/react-native/__tests__/ts/authentication/manage-email-password-users.test.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+};
+let higherScopedUser: Realm.User;
+// :remove-end:
+
+function AppWrapper() {
+  return (
+    <View>
+      <AppProvider id={APP_ID}>
+        <RegisterUser />
+        <UserProvider fallback={LogIn}>
```
Comment (takameyer) [inline examples/react-native/__tests__/ts/authentication/manage-email-password-users.test.tsx:24]:
```suggestion
        <UserProvider fallback={RegisterUser}>
```
I would set `RegisterUser` as the fallback and remove `LogIn`.

File: source/sdk/react-native/manage-users/manage-email-password-users.txt [hunk 4]
```diff
@@ -76,22 +87,8 @@ To confirm a registered user, pass a valid ``token`` and ``tokenId`` to
 :js-sdk:`EmailPasswordAuth.confirmUser()
 <Realm.Auth.EmailPasswordAuth.html#confirmUser>`.
```
Comment (takameyer) [inline source/sdk/react-native/manage-users/manage-email-password-users.txt:89]: It's not really clear to me from this code sample how I would confirm a user.  If this runs in React Native, where do the tokens come from?

**Review by takameyer — CHANGES_REQUESTED** [review]
I have a few requests

File: source/sdk/react-native/manage-users/manage-email-password-users.txt (hunk 4 above)
Comment (takameyer) [reply source/sdk/react-native/manage-users/manage-email-password-users.txt:89]: Good compromise.  Thanks!

**Review by takameyer — APPROVED** [review]
LGTM!

**Pull request merged**